internal/controller/clusterpolicy_controller.go

```diff
@@ -53,9 +54,8 @@ func (r *ClusterPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 	_ = log.FromContext(ctx)
 	_ = r.Log.WithValues("clusterpolicy", req.NamespacedName)
 
-	var clusterPolicy kyvernov1.ClusterPolicy
-
-	if err := r.Get(ctx, req.NamespacedName, &clusterPolicy); err != nil {
+	clusterPolicy, err := r.GetClusterPolicy(ctx, req.NamespacedName.Name)
+	if err != nil {
```

Comment:
I don't feel this is right because you are conditioning the "Get", and the controller should always fetch the ClusterPolicy in case it has changed.